projects/igniteui-angular/src/lib/grids/common/grid.interface.ts

```diff
@@ -707,3 +718,83 @@ export interface IgxGridMasterDetailContext {
 export interface IgxGroupByRowTemplateContext {
     $implicit: IGroupByRecord;
 }
+
+export interface IgxGridTemplateContext {
+    $implicit: GridType
+}
+
+export interface IgxGridRowTemplateContext {
+    $implicit: RowType
+}
+
+export interface IgxGridRowDragGhostContext {
+    $implicit: any, // this is the row data
+    data: any, // this is also the row data for some reason.
+    grid: GridType
+}
+
+export interface IgxGridEmptyTemplateContext {
+    $implicit: undefined
+}
+
+export interface IgxGridRowEditTemplateContext {
+    $implicit: undefined,
+    rowChangesCount: number,
+    endEdit:  (commit: boolean, event?: Event) => void
+}
+
+export interface IgxGridRowEditTextTemplateContext {
+    $implicit: number
+}
+
+export interface IgxGridRowEditActionsTemplateContext {
+    $implicit: (commit: boolean, event?: Event) => void
+}
+
+export interface IgxGridHeaderTemplateContext {
+    $implicit: HeaderType
+}
+
+export interface IgxColumnTemplateContext {
+    $implicit: ColumnType,
+    column: ColumnType
+}
+
+export interface IgxCellTemplateContext {
+    $implicit: any,
+    additionalTemplateContext: any,
+    formControl?: any,
+    defaultErrorTemplate?: any,
+    cell?: CellType
+}
+
+export interface IgxRowSelectorTemplateContext {
+    $implicit: {
+        index: number,
+        rowID: any,
+        selected: boolean,
+        select?: () => void,
+        deselect?: () => void
+    }
+}
+
+export interface IgxGroupByRowSelectorTemplateContext {
+    $implicit: {
+        selectedCount: number,
+        totalCount: number,
+        groupRow: RowType
```

Comment:
Shouldnt this be with type `IGroupByRecord`?  As long as I see in the `IgxGroupByRowSelectorDirective` we pass the groupRow